### ISSUE TITLE
GitHub Enterprise Server support?

### ISSUE THREAD

**Issue by sriram275** [integrations/microsoft-teams#27]
How about supporting signin with GitHub Enterprise Server?

**Comment by SonicGD** [comment]
I think there is another app for this - https://teams.microsoft.com/l/app/ca319335-400e-43c1-bca4-b6c745ce3093?source=store-copy-link

**Comment by ashokirla** [comment]
@sriram275 Yes, we have plans to support GHES as well. But, we will only pick it up early next year.

**Comment by klctrl** [comment]
@SonicGD Your referred app only offers notifications but lacks all the nice features of a deeper integration that this new app is about. (Wouldn't be surprised if this new app will at some point replace the one you mention above.)

**Comment by MagnusTim** [comment]
Hi,
Any news about when you will implement GHES support?

**Comment by byrneh** [comment]
Hi,
On the Team app aside of things, I was informed the new Git Hub app currently labelled “Preview” will will go full GA (Q1 2021) and that will be the only app you need to access Git Hub Cloud (online) and Git Hub Enterprise Server (on premise) , the GitHub Enterprise App will be retired.”

Is there any update you can provide in the README or in this issues on
1. The date this App integration will change from status "Public Preview" to GA?
2. When will Support for GHES be confirmed here and on what suggested date for preview/GA?, the [README](https://github.com/integrations/microsoft-teams/blob/master/Readme.md), its only listed under

**Comment by lolphert** [comment]
> @sriram275 Yes, we have plans to support GHES as well. But, we will only pick it up early next year.

@sriram275 Is there any update on this ? Does your team have a timeline for getting this completed 😄 ?

**Comment by paulanguiano** [comment]
We have also been sorely lacking this feature since adopting Teams.  Since the last update was six months ago, can we get any insight into the status and scheduling for this?

**Comment by hotgazpacho** [comment]
Bump. Any word on when the ability to use this with an on-prem Github Enterprise will be released?

**Comment by trackitsvalue** [comment]
Bump. The company I work for also uses Github Enterprise server and Teams.

**Comment by byrneh** [comment]
any update?

**Comment by BogdanAndreiRoman** [comment]
Bump. Any updated for on-prem integration?

**Comment by tjaroszuk** [comment]
Hello @ashokirla, Is there any chance for this feature to be added anytime soon? As you can see, there's a large user group looking forward to using this integration in their day-to-day work.

**Comment by icklers** [comment]
I give a clear +1 to GHES support, too, as we heavily use MS Teams and are currently moving our repos to GHES.

**Comment by gt3M** [comment]
+1

**Comment by bkottcamp** [comment]
+1

**Comment by Tschwen** [comment]
bump. Now it's nearly a year without any update here?

**Comment by kkankala** [comment]
bump.

**Comment by ashokirla** [comment]
Hi @hotgazpacho @SonicGD @bkottcamp @sriram275 @Tschwen @arndkeyz @Sony-Varghese @gt3M @icklers @kkankala @tjaroszuk @trackitsvalue @BogdanAndreiRoman @byrneh @paulanguiano @lolphert 👋 
Thanks for your patience. We are almost close to completing GHES integration. We are planning rollout as part of GHES 3.6.
This will be a private preview where we will enable the integration for interested customers. This will help in getting feedback before we announce GA.

If you are interested to be part of private preview, please reachout to me at chatop@github.com.

Thanks,
Ashok 🙏

**Comment by galvesribeiro** [comment]
Hello @ashokirla! Tried to send you an email but it bounced :(

<img width="638" alt="image" src="https://user-images.githubusercontent.com/4714040/171077199-70624049-1625-4143-b978-f018709abf6c.png">

**Comment by SauravNahata** [comment]
Hello @ashokirla, Any updates on when is this planned for GA? Thanks!

**Comment by ashokirla** [comment]
Hi all 👋 ,

We are planning to rollout GHES integration as part of GHES 3.6. We have completed the implementation and waiting for 3.6 release (early August). 

This will be a private preview, where interested customers can enrol for the preview. We will help you with the onboarding process. Will share more details once we are close to the release date. 

Those who are interested in private preview, please reach out to us at chatops@github.com. Your early feedback will help us in improving the app and GA announcement.

**Comment by jonbrohauge** [comment]
Sent an email, requesting to be signed up.
We expect to upgrade to 3.6.x early September.

**Comment by ashokirla** [comment]
@jonbrohauge Thanks for reaching out. I have responded on the email with more details and the next steps to tryout the integration.

**Comment by ashokirla** [comment]
> Hello @ashokirla! Tried to send you an email but it bounced :(
> 
> <img alt="image" width="638" src="https://user-images.githubusercontent.com/4714040/171077199-70624049-1625-4143-b978-f018709abf6c.png">
@galvesribeiro My apologies. There was a typo. Can you reachout to us at chatops@github.com

**Comment by SauravNahata** [comment]
Any feedbacks that can be shared from private preview! 🙂
Also any timelines on GA of the same? Thanks

**Comment by tdupoiron** [comment]
I sent chatops a couple of emails and I did not receive any feedback.
Do you know how to get this beta feature please ? I would like to onboard one of my customers. Thanks.

**Comment by tdupoiron** [comment]
Hi here. Now that 3.8 is out the release note mentions the integration with Teams.
But I don't see in Teams any update to choose for an Enterprise Server authentication. How does it work? Thank you.

[Release note:](https://docs.github.com/en/enterprise-server@3.8/admin/release-notes#3.8.0)

> [Integrations](https://docs.github.com/enterprise-server@3.8/admin/release-notes#integrations)
> Users can save time and switch context less often by receiving and acting on real-time updates about GitHub Enterprise Server activity directly within Slack or Microsoft Teams. GitHub’s integrations for these services are now generally available. For more information, see “[GitHub extensions and integrations](https://docs.github.com/en/enterprise-server@3.8/get-started/customizing-your-github-workflow/exploring-integrations/github-extensions-and-integrations).”
